frontend/src/libs/packages/map/map.package.ts

```diff
@@ -1,43 +1,66 @@
-import truckImg from '~/assets/img/tow-truck.png';
 import { ApplicationError } from '~/libs/exceptions/exceptions.js';
 
-import {
-  TRUCK_IMG_ANCHOR_X,
-  TRUCK_IMG_ANCHOR_Y,
-  TRUCK_IMG_HEIGHT,
-  TRUCK_IMG_WIDTH,
-} from './libs/constants/constants.js';
-import { rotateImg } from './libs/helpers/rotate-img.helper.js';
+import { MAP_INFO_WINDOW_WIDTH } from './libs/constants/constants.js';
+import { createIcon } from './libs/helpers/helpers.js';
 import { type IMapService } from './libs/interfaces/interfaces.js';
 import mapStyle from './map.config.json';
 
 type Constructor = {
   mapElement: HTMLDivElement;
   center?: google.maps.LatLngLiteral;
   zoom: number;
+  bounds?: google.maps.LatLngBounds;
 };
 
+// NOTE: The map is supposed to accept either the center or the bounds (if the center is unknown)
 class MapService implements IMapService {
   private map: google.maps.Map | null = null;
 
   private directionsService: google.maps.DirectionsService;
 
   private directionsRenderer: google.maps.DirectionsRenderer;
 
-  public constructor({ mapElement, center, zoom }: Constructor) {
+  private infoWindow: google.maps.InfoWindow;
+
+  private geoCoder: google.maps.Geocoder;
+
+  public constructor({ mapElement, center, zoom, bounds }: Constructor) {
     this.directionsService = new google.maps.DirectionsService();
     this.directionsRenderer = new google.maps.DirectionsRenderer({
       suppressMarkers: true,
     });
+    this.infoWindow = new google.maps.InfoWindow({
+      maxWidth: MAP_INFO_WINDOW_WIDTH,
+    });
+    this.geoCoder = new google.maps.Geocoder();
 
-    this.initMap(mapElement, center, zoom);
+    this.initMap({ mapElement, center, zoom, bounds });
   }
 
-  private initMap(
-    mapElement: HTMLDivElement,
-    center?: google.maps.LatLngLiteral,
-    zoom?: number,
-  ): void {
+  private initMap({
+    mapElement,
+    center,
+    zoom,
+    bounds,
+  }: {
+    mapElement: HTMLDivElement;
+    center?: google.maps.LatLngLiteral;
+    zoom?: number;
+    bounds?: google.maps.LatLngBounds;
+  }): void {
```

Comment:
You already have this type, it's called Constructor. You can reuse it (don't forget to rename, because Constructor will not suit anymore because of usage in multiple places).